### PR TITLE
chore: fix PyPi trusted publishing in GitHub Actions

### PR DESCRIPTION
**Pull request by clintval**
Closes: https://github.com/fulcrumgenomics/prymer/issues/3
Closes: https://github.com/fulcrumgenomics/prymer/issues/24

And the updated PyPi trusted publisher config:

Screenshot 2024-09-18 at 11 25 30 AM

### CODE REVIEW

**Comment by clintval** [comment]
This will not be possible to test without making a release of some kind.

**Comment by clintval** [comment]
Very confused why tests are failing...

File: .github/workflows/publish_prymer.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: publish
+name: publish prymer
```
Comment (msto) [inline .github/workflows/publish_prymer.yml:1]:
```suggestion
name: Publish prymer
```

**nit** I like to use title case so it renders a bit more nicely in the checks UI

**Review by msto — APPROVED** [review]
lgtm! thanks for fixing

File: .github/workflows/publish_prymer.yml (hunk 1 above)
Comment (clintval) [reply .github/workflows/publish_prymer.yml:1]: > **nit** I like to use title case so it renders a bit more nicely in the checks UI

I agree! In this case I'm matching the repo's pattern of having them be lowercase so this one doesn't appear out of place.

Screenshot 2024-09-19 at 8 34 49 AM

**Review by tfenne — APPROVED** [review]

**Pull request merged**